python/pyspark/sql/connect/functions/builtin.py

```diff
@@ -2394,22 +2394,31 @@ def unbase64(col: "ColumnOrName") -> Column:
 unbase64.__doc__ = pysparkfuncs.unbase64.__doc__
 
 
-def ltrim(col: "ColumnOrName") -> Column:
-    return _invoke_function_over_columns("ltrim", col)
+def ltrim(col: "ColumnOrName", trim: Optional["ColumnOrName"] = None) -> Column:
+    if trim is not None:
+        return _invoke_function_over_columns("ltrim", trim, col)
```

Comment:
it is kind of confusing on the arg order here:
1, connect: `_invoke_function_over_columns("ltrim", trim, col)` -> follow the order in constructor
https://github.com/apache/spark/blob/d85e7bc0beb49dd1d894d487cf6a5a02075280dd/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala#L1507

2, classic: `_invoke_function_over_columns("ltrim", col, trim)` follows the signature in scala functions
```
  def ltrim(e: Column, trimString: String): Column
  def ltrim(e: Column, trim: Column): Column
```

Reply:
probably we can unify the function invocations of connect and classic in some way